app/javascript/pages/components/gallery/2020/March.jsx

```diff
@@ -0,0 +1,89 @@
+/* eslint-disable max-len */
+import * as d3 from 'd3';
+import React, { useEffect } from 'react';
+import axios from 'axios';
+import D3Map from '../D3Map';
+
+const tooltipHtml = (development) => `<p>${development.attributes.name}</p>`;
+
+const tooltipLeft = (xCoordinate) => {
+  const tooltipWidth = +getComputedStyle(document.querySelector('.calendar-viz__tooltip')).width.slice(0, -2);
+  if (xCoordinate >= 300) {
+    return `${xCoordinate - tooltipWidth}px`;
+  }
+  return `${xCoordinate + 10}px`;
+};
+
+const tooltipTop = (yCoordinate) => {
+  const tooltipHeight = +getComputedStyle(document.querySelector('.calendar-viz__tooltip')).height.slice(0, -2);
+  if (yCoordinate >= 250) {
+    return `${yCoordinate - tooltipHeight - 10}px`;
+  }
+  return `${yCoordinate + 20}px`;
+};
+
+const addMapData = (massbuilds) => {
+  const projection = d3.geoAlbers()
+    .scale(37000)
+    .rotate([71.057, 0])
+    .center([0.33, 42.37])
+    .translate([960 / 2, 500 / 2]);
+
+  const tooltip = d3.select('.calendar-viz__tooltip');
+  const marchMap = d3.select('.calendar-viz__d3-map');
+
+  marchMap.append('g')
+    .attr('class', 'd3-map__points')
+    .selectAll('circle')
+    .data(massbuilds)
+    .enter()
+    .append('circle')
+    .attr('cx', (development) => projection(development.coordinates)[0])
+    .attr('cy', (development) => projection(development.coordinates)[1])
+    .attr('r', 3)
+    .attr('class', 'd3-map__point')
+    .attr('fill', 'red')
+    .attr('cursor', 'pointer')
+    .on('mousemove', (development) => {
+      tooltip.transition()
+        .duration(50)
+        .style('opacity', 0.9);
+      tooltip.html(tooltipHtml(development))
+        .style('left', tooltipLeft(projection(development.coordinates)[0]))
+        .style('top', tooltipTop(projection(development.coordinates)[1]));
+    })
+    .on('mouseleave', () => {
+      tooltip.transition()
+        .duration(200)
+        .style('opacity', 0);
+    });
+
+  d3.select('d3-map__points').raise();
+};
+
+const March = () => {
+  useEffect(() => {
+    const massBuildsUrl = 'https://api.massbuilds.com/developments?filter%5Bstatus%5D%5Bcol%5D=status&filter%5Bstatus%5D%5Bname%5D=Status&filter%5Bstatus%5D%5BglossaryKey%5D=STATUS&filter%5Bstatus%5D%5Btype%5D=string&filter%5Bstatus%5D%5Boptions%5D%5B%5D=completed&filter%5Bstatus%5D%5Boptions%5D%5B%5D=in_construction&filter%5Bstatus%5D%5Boptions%5D%5B%5D=planning&filter%5Bstatus%5D%5Boptions%5D%5B%5D=projected&filter%5Bstatus%5D%5Bfilter%5D=metric&filter%5Bstatus%5D%5Bvalue%5D=in_construction&filter%5Bstatus%5D%5Binflector%5D=eq&filter%5Byear_compl%5D%5Bcol%5D=year_compl&filter%5Byear_compl%5D%5Bname%5D=Year%20complete&filter%5Byear_compl%5D%5BglossaryKey%5D=YEAR_COMPLETE&filter%5Byear_compl%5D%5Btype%5D=number&filter%5Byear_compl%5D%5Bfilter%5D=metric&filter%5Byear_compl%5D%5Bvalue%5D=2018&filter%5Byear_compl%5D%5Binflector%5D=%3E';
```

Comment:
Not a big need or deal but if this URL might change in the future it might be worth refactoring into this style:
```
// Make a request for a user with a given ID
axios.get('/user?ID=12345')
// Optionally the request above could also be done as
axios.get('/user', {
    params: {
      ID: 12345
    }
  })
```

Reply:
I'm honestly not entirely sure how MassBuilds takes params, so I've just been making the appropriate query in the UI and then going into the network tab and grabbing the appropriate endpoint. I didn't realize that this syntax existed, though, so will definitely keep it in mind.